core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -450,16 +450,12 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
   }
 
   def updateGroupMembership(groupRef: RawlsGroupRef, addUsers: Set[RawlsUserRef] = Set.empty, removeUsers: Set[RawlsUserRef] = Set.empty, addSubGroups: Set[RawlsGroupRef] = Set.empty, removeSubGroups: Set[RawlsGroupRef] = Set.empty): Future[RawlsGroup] = {
-    updateGroupMembershipInternal(groupRef) { group =>
-      group.copy(
-        users = group.users ++ addUsers -- removeUsers,
-        subGroups = group.subGroups ++ addSubGroups -- removeSubGroups
-      )
-    }
-  }
 
-  def updateGroupMembershipPolicy(groupRef: RawlsGroupRef, addUsers: Set[RawlsUserRef] = Set.empty, removeUsers: Set[RawlsUserRef] = Set.empty, addSubGroups: Set[RawlsGroupRef] = Set.empty, removeSubGroups: Set[RawlsGroupRef] = Set.empty): Future[RawlsGroup] = {
-    updateGroupMembershipInternalPolicy(groupRef) { group =>
+    val groupOrPolicy = groupRef.groupName.value match {
+      case s if s.matches("([^@]+)@([^@]+)@([^@]+)") => updateGroupMembershipInternalPolicy(_)
```

Comment:
this pattern is in DirectorySubjectNameSupport.policyGroupNamePattern and should be reused somehow. Also look at how that pattern in used in other case statements for a little clean case statement.